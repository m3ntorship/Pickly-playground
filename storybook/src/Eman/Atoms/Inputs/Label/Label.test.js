import React from "react";
import renderer from "react-test-renderer";
import { Label } from "./Label";

describe("Testing Label", () => {
  it("Testing Label type Default", () => {
    const tree = renderer.create(<Label type="Default" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Label type Success", () => {
    const tree = renderer.create(<Label type="Success" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Label type Error", () => {
    const tree = renderer.create(<Label type="Error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Label type Disabled", () => {
    const tree = renderer.create(<Label type="Disabled" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Label with no props", () => {
    const tree = renderer.create(<Label />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
