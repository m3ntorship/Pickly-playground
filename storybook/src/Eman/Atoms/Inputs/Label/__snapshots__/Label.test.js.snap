// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Label Testing Label type Default 1`] = `
Array [
  <label
    className="block my-2 text-xs  text-dark-grey"
  />,
  <div
    className="
          flex items-center  w-full px-m py-xs relative  rounded-md  text-sm text-grey-shd1 hover:border-grey-shd2 focus:border-dark
          border border-grey-shd5"
  >
    <input
      className="w-full focus:outline-none focus:placeholder-dark focus:text-dark disabled:opacity-50"
      disabled={false}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
    
  </div>,
]
`;

exports[`Testing Label Testing Label type Disabled 1`] = `
Array [
  <label
    className="block my-2 text-xs  text-dark-grey"
  />,
  <div
    className="
          flex items-center  w-full px-m py-xs relative  rounded-md  text-sm text-grey-shd1 hover:border-grey-shd2 focus:border-dark
          border border-grey-shd7"
  >
    <input
      className="w-full focus:outline-none focus:placeholder-dark focus:text-dark disabled:opacity-50"
      disabled="disabled"
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
    
  </div>,
]
`;

exports[`Testing Label Testing Label type Error 1`] = `
Array [
  <label
    className="block my-2 text-xs  text-dark-grey"
  />,
  <div
    className="
          flex items-center  w-full px-m py-xs relative  rounded-md  text-sm text-grey-shd1 hover:border-grey-shd2 focus:border-dark
          border border-error"
  >
    <input
      className="w-full focus:outline-none focus:placeholder-dark focus:text-dark disabled:opacity-50"
      disabled={false}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
    <svg
      className="absolute right-3"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.96883 1.33337C4.31016 1.33337 1.3335 4.32404 1.3335 8.00004C1.3335 11.676 4.32416 14.6667 8.00016 14.6667C11.6762 14.6667 14.6668 11.676 14.6668 8.00004C14.6668 4.32404 11.6622 1.33337 7.96883 1.33337ZM8.00016 13.3334C5.0595 13.3334 2.66683 10.9407 2.66683 8.00004C2.66683 5.05937 5.04483 2.66671 7.96883 2.66671C10.9275 2.66671 13.3335 5.05937 13.3335 8.00004C13.3335 10.9407 10.9408 13.3334 8.00016 13.3334Z"
        fill="#FE544A"
      />
      <path
        d="M7.3335 4.66663H8.66683V9.33329H7.3335V4.66663ZM7.3335 9.99996H8.66683V11.3333H7.3335V9.99996Z"
        fill="#FE544A"
      />
    </svg>
  </div>,
]
`;

exports[`Testing Label Testing Label type Success 1`] = `
Array [
  <label
    className="block my-2 text-xs  text-dark-grey"
  />,
  <div
    className="
          flex items-center  w-full px-m py-xs relative  rounded-md  text-sm text-grey-shd1 hover:border-grey-shd2 focus:border-dark
          border border-success"
  >
    <input
      className="w-full focus:outline-none focus:placeholder-dark focus:text-dark disabled:opacity-50"
      disabled={false}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
    <svg
      className="absolute right-3"
      fill="none"
      height="9"
      viewBox="0 0 11 9"
      width="11"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.66681 6.39061L1.47148 4.19527L0.528809 5.13794L3.66681 8.27594L10.1381 1.8046L9.19547 0.861938L3.66681 6.39061Z"
        fill="#07B255"
      />
    </svg>
  </div>,
]
`;

exports[`Testing Label Testing Label with no props 1`] = `
Array [
  <label
    className="block my-2 text-xs  text-dark-grey"
  />,
  <div
    className="
          flex items-center  w-full px-m py-xs relative  rounded-md  text-sm text-grey-shd1 hover:border-grey-shd2 focus:border-dark
          border border-grey-shd7"
  >
    <input
      className="w-full focus:outline-none focus:placeholder-dark focus:text-dark disabled:opacity-50"
      disabled={false}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
    
  </div>,
]
`;
